ui/components/multichain/account-list-item-menu/account-list-item-menu.js

```diff
@@ -121,6 +125,14 @@ export const AccountListItemMenu = ({
     };
   }, [handleClickOutside]);
 
+  const handlePinningToggle = (address, pin) => {
+    const updatedPinnedAccountList = pin
+      ? pinnedAccountList.filter((item) => item !== address)
+      : [...pinnedAccountList, address];
+
+    dispatch(updateAccountsList(updatedPinnedAccountList));
+  };
```

Comment:
Should we add actions called `unpinAccount(address)` and `pinAccount(address)` instead of doing data management in the component?

Reply:
I don't think we need this pinning and unpinning function outside this component and the updateAccountsList is a global action. So, I think we can keep them together. Wdyt?

Reply:
I have separated both the actions here https://github.com/MetaMask/metamask-extension/pull/21865/commits/afa72376c7ad96cf4e1c533d134351ced95fbcff. I haven't added them separately in the actions.ts as both these functions are dispatching the global action updatedPinnedAccountList which updates the selector internally. If we move them to actions, I think it will be a dispatch inside a dispatch action.